File: Branching statement/Question1/question1.java
import java.util.Scanner;

public class question1 {

    static void SquareAndCude() {
        Scanner input = new Scanner(System.in);
        System.out.println("Enter a number: ");
        float number = input.nextFloat();
        System.out.println("Square of " + number + " is " + (number * number));
        System.out.println("Cube of " + number + " is " + (number * number * number));
    }

    static void Leap() {
        Scanner input = new Scanner(System.in);
        System.out.println("Enter a year: ");
        int year = input.nextInt();
        if (year % 4 == 0) {
            if (year % 100 == 0) {
                if (year % 400 == 0) {
                    System.out.println(year + " is a leap year");
                } else {
                    System.out.println(year + " is not a leap year");
                }
            } else {
                System.out.println(year + " is a leap year");
            }
        } else {
            System.out.println(year + " is not a leap year");
        }
    }

    public static void main(String[] args) {
        int choice;

        System.out.println("Choose an option: \n");
        System.out.println("1. Square and cube of a number\n");
        System.out.println("2. Leap year\n");
        System.out.println("3. Exit\n");
        Scanner input = new Scanner(System.in);
        choice = input.nextInt();
        if (choice == 1) {
            SquareAndCude();
        } else if (choice == 2) {
            Leap();
        } else if (choice == 3) {
            System.out.println("Exiting...");
            System.exit(0);
        } else {
            System.out.println("Invalid choice");
        }
    }
}

File: Branching statement/Question1/question1.cpp
#include <iostream>

int SquareAndCube()
{
    float num, square, cude;
    std::cout << "Enter a number: ";
    std::cin >> num;

    square = num * num;
    cude = num * num * num;

    std::cout << "Square of " << num << " is " << square << std::endl;
    std::cout << "Cube of " << num << " is " << cude << std::endl;
    return 0;
}

int Leap()
{
    int year;
    std::cout << "Enter a year: ";
    std::cin >> year;

    if (year % 4 == 0)
    {
        if (year % 100 == 0)
        {
            if (year % 400 == 0)
            {
                std::cout << year << " is a leap year" << std::endl;
            }
            else
            {
                std::cout << year << " is not a leap year" << std::endl;
            }
        }
        else
        {
            std::cout << year << " is a leap year" << std::endl;
        }
    }
    else
    {
        std::cout << year << " is not a leap year" << std::endl;
    }
    return 0;
}

int main()
{
    int choice;

    std::cout << "Choose an option: \n"
              << std::endl;
    std::cout << "1. Square and cube of a number\n"
              << std::endl;
    std::cout << "2. Leap year\n"
              << std::endl;
    std::cin >> choice;

    if (choice == 1)
    {
        SquareAndCube();
    }
    else if (choice == 2)
    {
        Leap();
    }
    else
    {
        std::cout << "Invalid choice" << std::endl;
    }

    return 0;
}